File: BinarySearch.hs
binarySearch' item list oldidx
           | cur > item = binarySearch' item fh oldidx
           | cur < item = binarySearch' item sh $ oldidx + idx
           | otherwise = oldidx + idx
           where 
                idx = ((length list) `div` 2)
                cur = list !! idx
                (fh, sh) = splitAt idx list

binarySearch item list = binarySearch' item list 0

main = print $ binarySearch 1 [1,2,3,4,5,6,7,8,9]

File: binary-search.py
def binary_search(array, n):
    cur = len(array) // 2
    if array[cur] == n:
        return cur
    elif array[cur] > n:
        return binary_search(array[:cur], n, doprint)
    else:
        return binary_search(array[(cur + 1):], n, doprint)


binary_search([1, 2, 3, 4, 5, 6, 7, 8, 9], 1)
